.konflux/scripts/render_snapshot.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+import datetime
+import json
+import os
+import re
+
+
+def parse_image_refs(image_refs):
+    return json.loads(image_refs)
+
+
+def validate_component(component):
+    assert (
+        component["name"] != ""
+        and component["containerImage"] != ""
+        and component["revision"] != ""
+        and component["repository"] != ""
+    ), "Component must have component name, ref, revision and repository set. Check container image labels."
+
+
+def determine_component_version_suffix(application):
+    match = re.search(r"acs-(?P<version>\d+-\d+)", application)
+    if match:
+        return match.group('version')
+    return ""
+
+
+def process_component(component, name_suffix):
+    validate_component(component)
+    if name_suffix != "":
+        name = f"{component['name']}{name_suffix}"
+    else:
+        name = component["name"]
```

Comment:
[nits]
1. Please rename `name_suffix` -> `version_suffix` for consistency.
2. If the suffix is empty string, you can safely append it, i.e. no need in `if/else`.

```suggestion
def process_component(component, version_suffix):
    validate_component(component)
    name = f"{component['name']}{version_suffix}"
```

3. With that, you can just inline the f-string in the "name" assignment below.